### ISSUE TITLE
Rule: Disallow `{{yield}}` only templates.

### ISSUE THREAD

**Issue by rwjblue** [ember-template-lint/ember-template-lint#29]
Prevent `{{yield}}` only templates. They are not required, and only increase the total template payload size...

**Comment by oriSomething** [comment]
are you sure they're totally unneeded?
i think in case you make components for styling purposes it's actually understood. if there is no component. the `{{yield}}` made automatically? if not, i'm not sure this rule should be on by default

**Comment by rwjblue** [comment]
I don't understand what you mean. If you have a component JS file, but do not need a template then the {{yield}} is not needed (it is default behavior). If you have no JS **and** no template other than {{yield}} you shouldn't be using a component invocation at all (use normal DOM, it is faster and much clearer).

**Comment by oriSomething** [comment]
i had a case that i created a button that had only `{{yield}}` in the template, that all the functionality was in JS, the `{{yield}}` was for the text on the button only.
also i know from a friend who works on some project that because of a technical need he had some components that their template is only `{{yield}}`.
maybe those are extreme cases. but i barely see this rule helpful, and more of annoying thing every time i'll create a new component.

**Comment by rwjblue** [comment]
The cases you listed would have worked properly with no template.

**Comment by oriSomething** [comment]
maybe, but it would damage the ease to extend the template if needed.
anyway, as i told, i don't think this rule can create gains as much as it would be annoying while creating a new component in the CLI while the server works

**Comment by mmun** [comment]
I think this rule is interesting. I think it's worth exploring without enabling by default.

**Comment by rwjblue** [comment]
@lifeart - I'm not sure I follow? Having a template with _only_ `{{yield}}` seems bad regardless of contextual components...

**Comment by lifeart** [comment]
ah, I mean `{{yield (hash someProp=someValue)}}`.

Regarding `{{yield}}` only components it can be an usecase for component with tag wrapper for some styling or native DOM bindings for internal content.

ex:
```js
{
   classNames: ['some', 'nice', 'ui', 'panel'],
   tagName: 'div'
}
```
```hbs
   {{my-panel}} my content inside nice panel {{my-panel}}
```

-------------------------

```js
{
   classNames: ['some', 'nice', 'ui', 'panel'],
   tagName: 'div',
   didRender() {
      this.element.querySelector('button').addEventListener();
   }
}
```
```hbs
   {{my-panel}} <button>some native action-binded item</button> {{my-panel}}
```

**Comment by rwjblue** [comment]
@lifeart - In your example:

```js
// app/components/nice-ui-panel.js

export default Ember.Component.extend({
   classNames: ['some', 'nice', 'ui', 'panel'],
   tagName: 'div',
});
```

Having a template like:

```hbs
{{yield}}
```

Is **the default behavior**, having no template at all will work _exactly_ the same.

**Comment by lifeart** [comment]
omg, I believe `no-yield` == no block rendering, even without templates

**Comment by mike-north** [comment]
We should consider that enabling this rule would result in `ember g component` having lint problems immediately upon generation. Historically, we've tried to ensure that `ember g <thing>` commands result in "no problems", including making sure the initially-generated tests are "born already passing".

**Comment by rwjblue** [comment]
Sure, but that just means `ember g component` would need to update what it emits, not that we shouldn't do the lint rule 😉

**Comment by mike-north** [comment]
> Sure, but that just means ember g component would need to update what it emits, not that we shouldn't do the lint rule 😉

Sounds good. I'm just looking to see that `ember g` outputs aren't "born with problems".

**Comment by rwjblue** [comment]
We are on the same page, I agree that the emitted blueprints should pass the recommended set of lint rules.
